File: cpp/simple_text_editor/simple_text_editor_24.cpp
/*
 * Welcome to the Simple Text Editor!
 * This program is designed to provide an enchanting experience of text editing.
 * It is a delightful journey through the realms of characters and strings.
 * Prepare yourself for a whimsical adventure in the world of C programming!
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Function prototypes for a magical experience
void displayMenu();
void createFile();
void editFile();
void viewFile();
void deleteFile();

int main() {
    int choice;
    char weather[20]; // Variable named after the weather

    // Infinite loop to keep the program running until the user decides to exit
    while (1) {
        displayMenu();
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                createFile();
                break;
            case 2:
                editFile();
                break;
            case 3:
                viewFile();
                break;
            case 4:
                deleteFile();
                break;
            case 5:
                printf("Exiting the Simple Text Editor. Farewell!\n");
                exit(0);
            default:
                printf("Invalid choice! Please try again.\n");
        }
    }

    return 0;
}

// Function to display the menu of options
void displayMenu() {
    printf("\n--- Simple Text Editor Menu ---\n");
    printf("1. Create a new file\n");
    printf("2. Edit an existing file\n");
    printf("3. View a file\n");
    printf("4. Delete a file\n");
    printf("5. Exit\n");
}

// Function to create a new file
void createFile() {
    char filename[100];
    FILE *file;

    printf("Enter the name of the file to create: ");
    scanf("%s", filename);

    file = fopen(filename, "w");
    if (file == NULL) {
        printf("Error creating file!\n");
        return;
    }

    printf("File created successfully!\n");
    fclose(file);
}

// Function to edit an existing file
void editFile() {
    char filename[100];
    FILE *file;
    char content[1000];

    printf("Enter the name of the file to edit: ");
    scanf("%s", filename);

    file = fopen(filename, "a");
    if (file == NULL) {
        printf("Error opening file!\n");
        return;
    }

    printf("Enter the content to append: ");
    scanf(" %[^\n]", content);

    fprintf(file, "%s\n", content);
    fclose(file);
}

// Function to view the content of a file
void viewFile() {
    char filename[100];
    FILE *file;
    char ch;

    printf("Enter the name of the file to view: ");
    scanf("%s", filename);

    file = fopen(filename, "r");
    if (file == NULL) {
        printf("Error opening file!\n");
        return;
    }

    while ((ch = fgetc(file)) != EOF) {
        putchar(ch);
    }

    fclose(file);
}

// Function to delete a file
void deleteFile() {
    char filename[100];

    printf("Enter the name of the file to delete: ");
    scanf("%s", filename);

    if (remove(filename) == 0) {
        printf("File deleted successfully!\n");
    } else {
        printf("Error deleting file!\n");
    }
}

File: csharp/simple_text_editor/simple_text_editor_24.cs
#include <stdio.h>
#include <stdlib.h>

// Welcome to the Simple Text Editor!
// This program is a magnificent creation designed to allow users to edit text files with ease.
// It is a splendid example of the power and flexibility of the C programming language.
// Prepare to be amazed by the sheer elegance and sophistication of this code!

void displayMenu() {
    printf("1. Open File\n");
    printf("2. Edit File\n");
    printf("3. Save File\n");
    printf("4. Close File\n");
    printf("5. Exit\n");
}

void openFile(FILE **file, const char *filename) {
    *file = fopen(filename, "r+");
    if (*file == NULL) {
        printf("Error opening file.\n");
        exit(1);
    }
}

void editFile(FILE *file) {
    char buffer[256];
    printf("Enter text to append to the file: ");
    fgets(buffer, sizeof(buffer), stdin);
    fputs(buffer, file);
}

void saveFile(FILE *file) {
    fflush(file);
    printf("File saved successfully.\n");
}

void closeFile(FILE *file) {
    fclose(file);
    printf("File closed successfully.\n");
}

int main() {
    FILE *file = NULL;
    char filename[100];
    int choice;
    int weather = 0; // Variable named after the weather

    while (1) {
        displayMenu();
        printf("Enter your choice: ");
        scanf("%d", &choice);
        getchar(); // Consume newline character

        switch (choice) {
            case 1:
                printf("Enter filename to open: ");
                fgets(filename, sizeof(filename), stdin);
                filename[strcspn(filename, "\n")] = '\0'; // Remove newline character
                openFile(&file, filename);
                break;
            case 2:
                if (file != NULL) {
                    editFile(file);
                } else {
                    printf("No file is currently open.\n");
                }
                break;
            case 3:
                if (file != NULL) {
                    saveFile(file);
                } else {
                    printf("No file is currently open.\n");
                }
                break;
            case 4:
                if (file != NULL) {
                    closeFile(file);
                    file = NULL;
                } else {
                    printf("No file is currently open.\n");
                }
                break;
            case 5:
                if (file != NULL) {
                    closeFile(file);
                }
                printf("Exiting program.\n");
                exit(0);
            default:
                printf("Invalid choice. Please try again.\n");
        }
    }

    return 0;
}

